### PR TITLE
[Bugfix] Avoid CPU round-trip for Qwen3 talker text queue

### PR DESCRIPTION
**Pull request by JingwenGu0829**
Benchmark shown below.
Avoiding CPU round-trip for Qwen3 talker text queue.
#485

### CODE REVIEW

**Comment by yxs** [comment]
Verified on (n=1088 c=16, `OMNI_PAD_TIMING=1`):

| | pre-fix (main) | post-fix |
|---|---:|---:|
| pad `.to()` avg/call | 150 μs | **6.2 μs** |
| pad `.to()` per req | 5.25 ms | 0.21 ms |
| n50 | 8 | 8 every bucket |
| WER corpus | 2.73 % | 2.78 % (1R noise) |
| latency_mean | 2.252 s (5R) | 2.245 s (1R) |

Sub-noise on bulk `latency_mean` as expected — wins are on critical-path CPU + TTFT/ITL.

File: sglang_omni/models/qwen3_omni/pending_text_queue.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+def _as_rows(tensor: torch.Tensor) -> torch.Tensor | None:
+    tensor = tensor.detach()
+    if tensor.numel() == 0:
+        return None
+    if tensor.dim() == 1:
+        return tensor.reshape(1, -1)
+    if tensor.dim() == 2:
+        return tensor
+    return tensor.reshape(-1, tensor.shape[-1])
```
Comment (Hayden727) [inline sglang_omni/models/qwen3_omni/pending_text_queue.py]: This helper is becoming the small queue contract between builders and the runner, so I would keep its shape semantics strict. Silently flattening any rank >2 means a future talker can accidentally pass `[batch, seq, hidden]` or `[seq, codebook, hidden]` and get interleaved FIFO rows instead of a clear model-side decision. Can we accept only `[hidden]` / `[rows, hidden]` here and require callers to flatten model-specific layouts explicitly?

File: sglang_omni/models/qwen3_omni/pending_text_queue.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        remaining = self.rows[self.cursor :]
+        rows = rows.to(device=remaining.device, dtype=remaining.dtype)
+        self.rows = torch.cat([remaining, rows], dim=0)
+        self.cursor = 0
+
+
+def coerce_pending_text_queue(value: object) -> PendingTextTensorQueue:
+    if isinstance(value, PendingTextTensorQueue):
+        return value
```
Comment (Hayden727) [inline sglang_omni/models/qwen3_omni/pending_text_queue.py]: Returning an existing `PendingTextTensorQueue` by reference gives callers shared mutable cursor state. The previous builder path copied list/deque inputs into a fresh deque, so two requests could not accidentally consume each other. Since `build_sglang_talker_request` is a reusable request helper, please add a cheap queue `copy()`/`clone()` that can share row storage but owns its own cursor, and use it when coercing an existing queue.

File: sglang_omni/models/qwen3_omni/request_builders.py [hunk 3]
```diff
@@ -424,7 +427,7 @@ def build_sglang_talker_request(
     talker_input_ids: torch.Tensor | list[int] | None = None,
     input_embeds_are_projected: bool = False,
     pending_text_queue: (
-        collections.deque[torch.Tensor] | list[torch.Tensor] | torch.Tensor | None
+        PendingTextTensorQueue | list[torch.Tensor] | torch.Tensor | None
```
Comment (Hayden727) [inline sglang_omni/models/qwen3_omni/request_builders.py]: The implementation still accepts legacy deque/iterable queues through `coerce_pending_text_queue`, and the tests still cover deque inputs, but the public type now advertises only `list`. To keep this request-builder API clean for existing schedulers and future models, please type this as `Iterable[torch.Tensor] | torch.Tensor | PendingTextTensorQueue | None` (or explicitly keep `collections.deque`) instead of narrowing it to `list`.

**Review by Hayden727 — CHANGES_REQUESTED** [review]
Thanks for moving the hot path off the CPU; the direction matches #485. I think the queue helper needs a little more API hardening before merge, because it is now the contract between request builders, scheduler state, and the talker runner. The perf change should stay, but the public-ish semantics should stay explicit for future talker models.

**Review by Hayden727 — APPROVED** [review]
Rechecked e22267f. The latest commit addresses my previous review points: rank handling is now explicit, coercing an existing queue gives an independent cursor, and the request-builder type keeps the iterable/deque-compatible API surface. I only did static review plus py_compile locally; pytest is blocked in this local env because torch is not installed.
